src/main/java/org/breedinginsight/services/SampleSubmissionService.java

```diff
@@ -427,4 +427,28 @@ public Optional<SampleSubmission> updateSubmissionStatus(Program program, UUID s
 
         return submissionOptional;
     }
+
+    /**
+     * Deletes BrAPI plates and submission objects as well as sample submission record in bidb
+     * We do not currently cache plates or samples so don't need to worry about that
+     * @param submissionId sample submission UUID to delete
+     * @exception ApiException if a BrAPI call fails
+     */
+    public void deleteSampleSubmission(Program program, UUID submissionId) throws ApiException {
+        // create a batch of sampleIds and plateIds to delete
+        // get samples with the sample submission xref
+        List<BrAPISample> samples = sampleDAO.readSamplesBySubmissionIds(program, List.of(submissionId.toString()));
+
+        // extract sampleDbIds and plateDbIds to include in batches
+        List<String> sampleDbIds = samples.stream().map(BrAPISample::getSampleDbId).collect(Collectors.toList());
+        List<String> platesDbIds = samples.stream().map(BrAPISample::getPlateDbId).collect(Collectors.toList());
```

Comment:
```suggestion
        List<String> sampleDbIds = samples.stream().map(BrAPISample::getSampleDbId).distinct().collect(Collectors.toList());
        List<String> platesDbIds = samples.stream().map(BrAPISample::getPlateDbId).distinct().collect(Collectors.toList());
```

If you have multiple samples per plate, you end up with duplication in the list of plateDbIds 
([test file for example](https://github.com/user-attachments/files/18774949/bi_sample_submission_template_v01.2.xls)).
 I'm not sure it's possible to get duplication in the sampleDbIds, but I don't think it hurts to deduplicate both lists.

Reply:
Thanks! Comitted